include/native/parallel_execution_native.h

```diff
@@ -953,35 +1002,35 @@ void parallel_execution_native::do_pipeline(
 
   using input_item_type = typename Queue::value_type;
   using input_item_value_type = typename input_item_type::first_type::value_type;
-  using transform_result_type = 
-      decay_t<typename result_of<FarmTransformer(input_item_value_type)>::type>;
-  using output_item_value_type = experimental::optional<transform_result_type>;
-  using output_item_type = pair<output_item_value_type,long>;
 
-  auto output_queue = make_queue<output_item_type>();
+  using output_type = typename get_return<input_item_value_type, FarmTransformer>::type;
+  using output_optional_type = experimental::optional<output_type>;
+  using output_item_type = pair <output_optional_type, long> ;
+
+  using queue_type = mpmc_queue<output_item_type>;
+  using conditional_queue_type = 
+    typename std::conditional<check_queue_type<queue_type&, OtherTransformers...>(), queue_type&, queue_type>::type;
+  conditional_queue_type output_queue = 
```

Comment:
Why not auto?

Reply:
It's necessary because the function returns, depending on the input type, an object or a reference to the object. Since the auto is resolved as the object's type, it calls to the copy constructor. The way to solve this issue is to use 'auto&' as type, but for the function that returns the object is not valid.